repair_scrap/wizards/repair_make_scrap.py

```diff
@@ -0,0 +1,113 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class RepairMakeScrap(models.TransientModel):
+    _name = "repair_make_scrap.wizard"
+    _description = "Wizard to create scrap from repair"
+
+    item_ids = fields.One2many(
+        comodel_name="repair_make_scrap_item.wizard",
+        inverse_name="wiz_id",
+        string="Items",
+    )
+
+    @api.returns("repair.order")
+    def _prepare_item(self, line):
+        values = {
+            "product_id": line.product_id.id,
+            "product_qty": line.product_qty,
+            "location_id": line.location_id.id,
+            "scrap_location_id": line.repair_type_id.scrap_location_id.id,
+            "uom_id": line.product_id.uom_id.id,
+            "repair_id": line.id,
+            "lot_id": line.lot_id.id,
+        }
+        return values
+
+    @api.model
+    def default_get(self, fields_list):
+        context = self._context.copy()
+        res = super(RepairMakeScrap, self).default_get(fields_list)
+        repair_obj = self.env["repair.order"]
+        repair_ids = self.env.context["active_ids"] or []
+        active_model = self.env.context["active_model"]
+
+        if not repair_ids:
+            return res
+        assert active_model == "repair.order", "Bad context propagation"
+
+        items = []
+        lines = repair_obj.browse(repair_ids)
+        for line in lines:
+            items.append([0, 0, self._prepare_item(line)])
+        res["item_ids"] = items
+        context.update({"items_ids": items})
+        return res
+
+    def _create_scrap(self):
+        scraps = []
+        for item in self.item_ids:
+            repair = item.repair_id
+            if repair.state == "draft" or repair.state == "cancel":
+                raise ValidationError(_("Repair %s is not confirmed") % repair.name)
+            scrap = self._prepare_scrap(item)
+            scraps.append(scrap)
+            item.repair_id.scrap_ids |= scrap
+        return scraps
+
+    def action_create_scrap(self):
+        self._create_scrap()
+        return self.item_ids[0].repair_id.action_view_scrap_transfers()
```

Comment:
It may be dummy, bit if for some reason the user deletes all the lines there is an error here
```suggestion
        if self.item_ids:
            return self.item_ids[0].repair_id.action_view_scrap_transfers()
```